app/src/main/java/com/xmartlabs/template/controller/DbFlowController.java

```diff
@@ -0,0 +1,118 @@
+package com.xmartlabs.template.controller;
+
+import android.support.annotation.CheckResult;
+import android.support.annotation.NonNull;
+
+import com.annimon.stream.Stream;
+import com.raizlabs.android.dbflow.config.FlowManager;
+import com.raizlabs.android.dbflow.sql.language.Delete;
+import com.raizlabs.android.dbflow.sql.language.SQLCondition;
+import com.raizlabs.android.dbflow.sql.language.SQLite;
+import com.raizlabs.android.dbflow.sql.language.Select;
+import com.raizlabs.android.dbflow.sql.language.property.Property;
+import com.raizlabs.android.dbflow.structure.BaseModel;
+import com.xmartlabs.template.database.AppDataBase;
+import com.xmartlabs.template.model.DatabaseModel;
+
+import java.util.List;
+
+import io.reactivex.Completable;
+import io.reactivex.Maybe;
+import io.reactivex.Single;
+import io.reactivex.schedulers.Schedulers;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class DbFlowController<Id, D extends DatabaseModel<Id>> extends Controller
+    implements DatabaseProvider<Id, D, SQLCondition> {
+  @Getter(AccessLevel.PRIVATE)
+  final Class<D> modelClass;
+  @Getter(AccessLevel.PRIVATE)
+  final Property<Id> idProperty;
+
+  @CheckResult
+  @NonNull
+  @Override
+  public Single<List<D>> getEntities(@NonNull SQLCondition... conditions) {
+    return Single
+        .fromCallable(() ->
+            SQLite.select()
+                .from(getModelClass())
+                .where(conditions)
+                .queryList()
+        )
+        .subscribeOn(Schedulers.io());
+  }
+
+  @CheckResult
+  @NonNull
+  @Override
+  public Single<List<D>> updateEntities(List<D> newEntities, @NonNull SQLCondition... conditions) {
+    return Single
+        .fromCallable(() -> {
+          FlowManager.getDatabase(AppDataBase.class)
+              .executeTransaction(databaseWrapper -> {
+                SQLite.delete(getModelClass())
```

Comment:
I guess that deleting the entities before saving them could leak into some issues. For example, if you have a foreign key in other entity then the deletion may fail. If it doesn't fail (I think it could depend on the DB implementation) then the related entity may have an invalid ID, if it is autogenerated.

So, is the deletion before saving needed? Could not simply update the given entity?

Thoughts?